### PR TITLE
phpstan-8 for category class

### PR DESCRIPTION
**Pull request by ColonelMoutarde**
Changes proposed in this pull request:

- phpstan-8 for category class

Pull request checklist:

- [x] clear commit messages
- [x] code manually tested

### CODE REVIEW

File: app/Models/Category.php [hunk 1]
```diff
@@ -22,7 +22,7 @@ class FreshRSS_Category extends Minz_Model {
 	private $nbFeeds = -1;
 	/** @var int */
 	private $nbNotRead = -1;
-	/** @var array<FreshRSS_Feed>|null */
+	/** @var array<FreshRSS_Feed> */
```
Comment (Alkarex) [inline app/Models/Category.php]: I am not sure this is a good solution, as we need to test for nullability below: `if ($this->feeds === null)`

Comment (ColonelMoutarde) [reply app/Models/Category.php]: got to matermost ?

Comment (ColonelMoutarde) [reply app/Models/Category.php]: $feedDAO->listByCategory($this->id()); return always a array() of FreshRSS_Feed

File: app/Models/Category.php [hunk 2]
```diff
@@ -99,7 +99,12 @@ public function nbNotRead(): int {
...
 
-	/** @return array<FreshRSS_Feed> */
+	/**
+	 * @return array<FreshRSS_Feed>
+	 * @throws Minz_ConfigurationNamespaceException
+	 * @throws Minz_PDOConnectionException
+	 */
+
 	public function feeds(): array {
 		if ($this->feeds === null) {
```
Comment (Alkarex) [inline app/Models/Category.php:108]: This is the line testing for null

File: app/Models/Category.php (hunk 1 above)
Comment (Alkarex) [reply app/Models/Category.php]: See below, line 109: `if ($this->feeds === null)`

Comment (Alkarex) [reply app/Models/Category.php]: https://github.com/FreshRSS/FreshRSS/pull/5434/commits/2a0cbaaf03a4f56031ded209f8920c3322dc513c

**Review by Alkarex — APPROVED** [review]
